File: src/test/java/Pages/EngineeringGamesFeedback.java
package Pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

public class EngineeringGamesFeedback extends BasePage{
    public EngineeringGamesFeedback(WebDriver driver) {
        super(driver);
    }

    public void navigateSite() {
        driver.get("http://localhost:3000");
    }
    public WebElement clickInputNameField() {
        return driver.findElement(By.xpath("//input[@type='text']"));
    }
    public WebElement clickInputName() {
        return driver.findElement(By.xpath("//input[@type='text']"));
    }
    public WebElement clickSelectLanguage() {
        return driver.findElement(By.xpath("//select[@name='language']"));
    }
    public WebElement selectLanguage() {
        return driver.findElement(By.xpath("//option[@value='python']"));
    }
    public WebElement clickSelectLevel() {
        return driver.findElement(By.xpath("//select[@id='level']"));
    }

    public WebElement selectLevel() {
        return driver.findElement(By.xpath("//option[@value='junior']"));
    }


    public WebElement clickSubmitButton() {
        return driver.findElement(By.xpath("//button[@type='submit']"));
    }
    public WebElement clickTrueAnswerFirst() {
        return driver.findElement(By.xpath("//button[@value='3']"));
    }
    public WebElement clickTrueAnswerSecond() {
        return driver.findElement(By.xpath("//button[@value='0']"));
    }
    public WebElement clickReturnHomePage() {
        return driver.findElement(By.xpath("//button[@class='returnPage']"));
    }
    public WebElement clickSentFeedback() {
        return driver.findElement(By.xpath("//a[@class='btn'][4]"));
    }
    public WebElement clickNameField() {
        return driver.findElement(By.xpath("//input[@name='name']"));
    }
    public WebElement typeName() {
        return driver.findElement(By.xpath("//input[@name='name']"));
    }
//    public WebElement clickEmailField() {
//        return driver.findElement(By.xpath("//input[@name='email']"));
//    }
    public WebElement typeEmail() {
        return driver.findElement(By.xpath("//input[@name='email']"));
    }
//    public WebElement clickMessageField() {
//        return driver.findElement(By.xpath("//textarea[@name='message']"));
//    }
    public WebElement typeMessage() {
        return driver.findElement(By.xpath("//textarea[@name='message']"));
    }
    public WebElement clickViewFeedback() {
        return driver.findElement(By.xpath("//button[@id='viewFeedbacks']"));
    }
    public WebElement clickSendFeedback() {
        return driver.findElement(By.xpath("//button[@type='submit']"));
    }
}

File: cypress/support/page_objects/engineeringGamesFeedback.js
// cypress/pages/engineeringGamesFeedback.js

class EngineeringGamesFeedback {
    // Метод для переходу на сайт
    navigateSite() {
      cy.visit('http://localhost:3001');
    }
  
    // Методи для взаємодії з полями форми
    clickInputNameField() {
      return cy.get('input[type="text"]');
    }
  
    clickInputName() {
      return this.clickInputNameField();
    }
  
    clickSelectLanguage() {
      return cy.get('select[name="language"]');
    }
  
    selectLanguage() {
      return this.clickSelectLanguage().select('python');
    }
  
    clickSelectLevel() {
      return cy.get('select[id="level"]');
    }
  
    selectLevel() {
      return this.clickSelectLevel().select('junior');
    }
  
    clickSubmitButton() {
      return cy.get('button[type="submit"]');
    }
  
    clickTrueAnswerFirst() {
      return cy.get('button[value="0"]');
    }
  
    clickTrueAnswerSecond() {
      return cy.get('button[value="0"]');
    }
  
    clickReturnHomePage() {
      return cy.get('button.returnPage');
    }
  
    clickSentFeedback() {
      return cy.get('a.btn').eq(3); // Використовуємо eq для вибору 4-го елемента
    }
  
    clickNameField() {
      return cy.get('input[name="name"]');
    }
  
    typeName() {
      return this.clickNameField();
    }
  
    clickEmailField() {
      return cy.get('input[name="email"]');
    }
  
    typeEmail() {
      return this.clickEmailField();
    }
  
    clickMessageField() {
      return cy.get('textarea[name="message"]');
    }
  
    typeMessage() {
      return this.clickMessageField();
    }
  
    clickViewFeedback() {
      return cy.get('button[id="viewFeedbacks"]');
    }
  
    clickSendFeedback() {
      return this.clickSubmitButton();
    }
  }
  
  export default EngineeringGamesFeedback;
  